### ISSUE TITLE
Vite plugin no longer works with `vite-tsconfig-paths` in v4

### ISSUE THREAD

**Issue by jakst** [vanilla-extract-css/vanilla-extract#1295]
### Describe the bug

Trying to upgrade to v4, and vanilla extract can no longer resolve aliased imports from [vite-tsconfig-paths](https://www.npmjs.com/package/vite-tsconfig-paths).

tsconfig.json
```jsonc
{
  // ...
  "compilerOptions": {
    // ...
    "paths": {
      "@/*": ["./src"],
    },
  },
}
```

Now imports like this no longer work:
```ts
import { vars } from '@/lib/vars.css' // File location 'src/lib/vars.css'
```

Error message
```
Failed to resolve module specifier "@/lib/vars.css". Relative references must start with either "/", "./", or "../".
```

To reproduce, clone the linked repo and run `pnpm install && pnpm build`.

### Reproduction

https://github.com/jakst/vanilla-extract-v4-ts-import-alias

### System Info

```shell
System:
    OS: macOS 14.2
    CPU: (12) arm64 Apple M3 Pro
    Memory: 49.50 MB / 36.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.18.2 - ~/.proto/shims/node
    Yarn: 4.0.2 - ~/.proto/shims/yarn
    npm: 9.8.1 - ~/.proto/shims/npm
    pnpm: 8.15.0 - ~/.proto/shims/pnpm
    bun: 1.0.25 - ~/.proto/shims/bun
    Watchman: 2024.01.22.00 - /opt/homebrew/bin/watchman
  Browsers:
    Brave Browser: 120.1.61.109
    Chrome: 121.0.6167.85
    Edge: 121.0.2277.83
    Safari: 17.2
  npmPackages:
    @vanilla-extract/css: ^1.14.1 => 1.14.1
    @vanilla-extract/vite-plugin: ^4.0.0 => 4.0.0
    vite: ^5.0.8 => 5.0.12
```

### Used Package Manager

pnpm

### Logs

```shell
[vanilla-extract] Failed to load url @/lib/vars.css (resolved id: @/lib/vars.css) in ~/code/tmp/vanilla-extract-v4/src/style.css.ts. Does the file exist?
```

### Validations

- [X] Check that there isn't [already an issue](https://github.com/vanilla-extract-css/vanilla-extract/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by nguyenkhois** [comment]
Have you tried using `resolve.alias` in `vite.config.ts`?

```
import { resolve } from 'path';
...

export default defineConfig({
	plugins: [tsconfigPaths(), vanillaExtractPlugin(), react()],
	resolve: {
		alias: {
			'@/': resolve(__dirname, 'src') + '/',
		}
	}
})
```

**Comment by f4-miyata** [comment]
(After updating to v4, aliases within url() started to resolve, but conversely, aliases within import stopped resolving. 
[(discussions)](https://github.com/vanilla-extract-css/vanilla-extract/discussions/1288#discussioncomment-8268104))

Even when resolve.alias is properly specified in vite.config.ts, aliases within import stopped resolving.

**Comment by mrm007** [comment]
Thanks for reporting. I have a fix in #1297.

Can you confirm it works when you install this snapshot?

```
pnpm add @vanilla-extract/vite-plugin@vite-plugin-resolve-plugins
```

**Comment by jakst** [comment]
Yeah, that version works @mrm007 👍
